app.rb

```diff
@@ -67,13 +67,20 @@ def self.report
             term_id = m.document[:legislative_period_id]
             terms[term_id][:overall][gender] += 1
             terms[term_id][:overall][:total] += 1
-            party = m.on_behalf_of_id
-            terms[term_id][:parties][party][:name] = party_id_to_name[party]
+            group_id = m.on_behalf_of_id
+            group = l.popolo.organizations.find { |o| o.id == group_id }
+            wikidata_identifier = ((group[:identifiers] || {}).find { |i| i[:scheme] == 'wikidata' } || {})[:identifier]
+            if wikidata_identifier
+              party = wikidata_identifier
+            else
+              party = group_id
+            end
+            terms[term_id][:parties][party][:name] = party_id_to_name[group_id]
```

Comment:
I'm finding it a little difficult to follow the logic here about which Party name is going to be picked under which conditions, when multiple parties are being combined like this, and certainly wouldn't be confident that it's doing the right thing, or that any future work in this area wasn't going to cause unexpected breakage. Even a very cursory set of tests here would be very useful, I think.